mcs/class/Mono.Posix/Mono.Unix.Android/AndroidUtils.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Reflection;
+
+// using Android.Runtime;
+// using Android.Util;
+
+namespace Mono.Unix.Android
+{
+	internal sealed class AndroidUtils
+	{
+		const string TAG = "Mono.Posix";
+
+		delegate void DetectCPUAndArchitecture (out ushort builtForCPU, out ushort runningOnCPU, out bool is64bit);
+
+		readonly static DetectCPUAndArchitecture detectCPUAndArchitecture;
+
+		static AndroidUtils ()
+		{
+			Type androidRuntime = Type.GetType ("Android.Runtime.AndroidEnvironment, Mono.Android", true);
+
+			MethodInfo mi = androidRuntime.GetMethod ("DetectCPUAndArchitecture", BindingFlags.NonPublic | BindingFlags.Static);
+			detectCPUAndArchitecture = (DetectCPUAndArchitecture) Delegate.CreateDelegate (typeof(DetectCPUAndArchitecture), mi);
+		}
+
+		public static bool AreRealTimeSignalsSafe ()
```

Comment:
IDK, as I said before, I'm bringing code from XA. That's a question for @jonpryor